src/coreclr/src/jit/flowgraph.cpp

```diff
@@ -453,6 +450,15 @@ void Compiler::fgEnsureFirstBBisScratch()
             block->inheritWeight(fgFirstBB);
         }
 
+        // The first block has an implicit ref count which we must
+        // remove. Note the ref count could be greater that one, if
+        // the first block is not scratch and is targeted by a
+        // branch.
+        assert(fgFirstBB->bbRefs >= 1);
+        fgFirstBB->bbRefs--;
```

Comment:
Since you're here, it works be nice to update the header, including when and under what pre-conditions it will be called.

Reply:
Will do.